src/main/java/in/projecteka/gateway/SecurityConfiguration.java

```diff
@@ -0,0 +1,181 @@
+package in.projecteka.gateway;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jwt.proc.ConfigurableJWTProcessor;
+import in.projecteka.gateway.clients.ClientRegistryProperties;
+import in.projecteka.gateway.clients.common.Authenticator;
+import in.projecteka.gateway.clients.common.CentralRegistryTokenVerifier;
+import in.projecteka.gateway.clients.common.UserAuthenticator;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.ReactiveAuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity;
+import org.springframework.security.config.web.server.ServerHttpSecurity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextImpl;
+import org.springframework.security.web.server.SecurityWebFilterChain;
+import org.springframework.security.web.server.context.ServerSecurityContextRepository;
+import org.springframework.util.AntPathMatcher;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.net.URL;
+import java.security.SecureRandom;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Configuration
+@EnableWebFluxSecurity
+public class SecurityConfiguration {
+
+    private static final List<Map.Entry<String, HttpMethod>> SERVICE_ONLY_URLS = new ArrayList<>() {
+        {
+
+        }
+    };
+
+    @Bean
+    public SecurityWebFilterChain securityWebFilterChain(
+            ServerHttpSecurity httpSecurity,
+            ReactiveAuthenticationManager authenticationManager,
+            ServerSecurityContextRepository securityContextRepository) {
+        final String[] WHITELISTED_URLS = {"/**.json",
+                "/ValueSet/**.json",
+                "/**.html",
```

Comment:
actually none of them are required. They are introduced because we wanted to serve the Swagger API but we can introduce them as and when required